src/pal/src/include/pal/stackstring.hpp

```diff
@@ -23,17 +23,25 @@ class StackString
 
     void DeleteBuffer()
     {
-        if (m_innerBuffer != m_buffer)
+        if ( m_innerBuffer != m_buffer)
             PAL_free(m_buffer);
 
         m_buffer = NULL;
         return;
     }
 
-    void ReallocateBuffer(SIZE_T count)
+    void ReallocateBuffer(SIZE_T count, BOOL preserve)
     {
         // count is always > STACKCOUNT here.
-        T * newBuffer = (T *)PAL_malloc((count + 1) * sizeof(T));
+        // We got so far, we will allocate a little extra
+        // to prevent frequent allocations
+#if _DEBUG
+        SIZE_T count_allocated = count;
```

Comment:
Expecting that 100 bytes would be sufficient for the case you've mentioned is kind of a lotery. It will work as you say in some cases, but it would not be enough in others. 
But let's not spend more time on this. Keep it here if you wish. But please modify the reservation calls to account for the small things too.